File: src/test/java/com/d20charactersheet/framework/boc/service/DnDv35RuleServiceFeatTest.java
package com.d20charactersheet.framework.boc.service;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import com.d20charactersheet.framework.boc.model.Character;
import com.d20charactersheet.framework.boc.model.ClassLevel;

public class DnDv35RuleServiceFeatTest extends DnDv35RuleServiceTestCase {

  @Test
  public void testGetNumberOfFeats() throws Exception {
    final Character character = createCharacter(new ClassLevel(wizard, 5), elf);
    int numberOfFeats = ruleService.getNumberOfFeats(character);
    assertEquals(2, numberOfFeats);

    character.setRace(human);
    numberOfFeats = ruleService.getNumberOfFeats(character);
    assertEquals(3, numberOfFeats);
  }

  @Test
  public void testGetNumberOfFeatsMultiClass() throws Exception {
    final Character character = createCharacter(new ClassLevel(wizard, 5), new ClassLevel(fighter, 2), human, 10);
    final int numberOfFeats = ruleService.getNumberOfFeats(character);
    assertEquals(4, numberOfFeats);
  }

}

File: src/test/java/com/d20charactersheet/framework/boc/service/DnDv35RuleServiceFeatTest.kt
package com.d20charactersheet.framework.boc.service

import com.d20charactersheet.framework.boc.model.ClassLevel
import org.junit.Assert
import org.junit.jupiter.api.Test

class DnDv35RuleServiceFeatTest : DnDv35RuleServiceTestCase() {

    @Test
    fun testGetNumberOfFeats() {
        val character = createCharacter(ClassLevel(wizard, 5), elf)
        var numberOfFeats = ruleService.getNumberOfFeats(character)
        Assert.assertEquals(2, numberOfFeats.toLong())
        character.race = human
        numberOfFeats = ruleService.getNumberOfFeats(character)
        Assert.assertEquals(3, numberOfFeats.toLong())
    }

    @Test
    fun testGetNumberOfFeatsMultiClass() {
        val character = createCharacter(ClassLevel(wizard, 5), ClassLevel(fighter, 2), human, 10)
        val numberOfFeats = ruleService.getNumberOfFeats(character)
        Assert.assertEquals(4, numberOfFeats.toLong())
    }
}